### ISSUE TITLE
Make an example notebook displaying the different kind of loss functions

### ISSUE THREAD

**Issue by jackbaker1001** [XanaduAI/GradDFT#29]
I want a notebook where we test using cost functions that:

(1) Train only the energy in a non self consistent way

(2) Train only the energy in a self-consistent way

(3) Train the energy and density in a way which enforces self consistency

The example will be for training on the binding curve of H_2 and testing on the binding curve of LiH. I also wish to generate a gif which shows the reduction of error in the LiH density at some bond length as a function of training epochs on H2. This will make a cool .gif animation for the repo README.md
